ably/realtime_conn_spec_test.go

```diff
@@ -2144,3 +2144,217 @@ func TestRealtimeConn_RTN2g(t *testing.T) {
 		t.Errorf("expected %q got %q", proto.LibraryString, lib)
 	}
 }
+
+func TestRealtimeConn_RTN19b(t *testing.T) {
+	t.Parallel()
+	connIDs := make(chan string)
+	var breakConn func()
+	var out, in chan *proto.ProtocolMessage
+	c, err := ably.NewRealtime(
+		// ably.WithLogLevel(ably.LogDebug),
+		ably.WithAutoConnect(false),
+		ably.WithKey("fake:key"),
+		ably.WithDial(func(protocol string, u *url.URL, timeout time.Duration) (proto.Conn, error) {
+			in = make(chan *proto.ProtocolMessage, 1)
+			in <- &proto.ProtocolMessage{
+				Action:       proto.ActionConnected,
+				ConnectionID: <-connIDs,
+				ConnectionDetails: &proto.ConnectionDetails{
+					ConnectionKey: "key",
+				},
+			}
+			out = make(chan *proto.ProtocolMessage, 16)
+			breakConn = func() { close(in) }
+			return ablytest.MessagePipe(in, out)(protocol, u, timeout)
+		}),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	changes := make(ably.ConnStateChanges, 2)
+	off := c.Connection.On(ably.ConnectionEventConnected, changes.Receive)
+	defer off()
+	c.Connect()
+	connIDs <- "1"
+	ablytest.Soon.Recv(t, nil, changes, t.Fatalf)
+	attaching := c.Channels.Get("attaching")
+	go attaching.Attach(context.Background())
```

Comment:
As commented in `sendAndWait`, AFAICT this `Attach` will block forever. If, instead of just `go`ing, you checked that ultimately those operations returned, this test would've caught that bug.

As a general pattern, `go` in tests should never happend and we should use `ablytest.ResultGroup.GoAdd` instead, in order to ensure we don't have goroutine leaks.

Reply:
done c47d51b